open_discussions/authentication.py

```diff
@@ -0,0 +1,92 @@
+"""Custom authentication for DRF"""
+import logging
+from base64 import b64decode, b64encode
+
+from django.conf import settings
+from django.contrib.auth import get_user_model
+from django.core import signing
+from rest_framework.authentication import BaseAuthentication
+
+User = get_user_model()
+
+HEADER_PREFIX = "Token "
+HEADER_PREFIX_LENGTH = len(HEADER_PREFIX)
+
+logger = logging.getLogger()
+
+
+def get_encoded_and_signed_subscription_token(user):
+    """
+    Returns a signed and encoded token for subscription authentication_classes
+
+    Args:
+        user (User): user to generate the token for
+
+    Returns:
+        str: a signed and encoded token for subscription authentication_classes
+    """
+    signer = signing.TimestampSigner()
+    signed_value = signer.sign(user.username)
+    return b64encode(signed_value.encode('utf-8')).decode('ascii')
+
+
+def decode_and_verify_username_from_token(token):
+    """
+    Returns the decoded username from a subscription token
+
+    Args:
+        token (str): the encoded token
+
+    Returns:
+        str: the decoded username
+    """
+
+    # we base64 encode this so it is urlsafe
+    try:
+        signed_value = b64decode(token)
+    except:  # pylint: disable=bare-except
+        return None
+
+    signer = signing.TimestampSigner()
+    try:
+        print(signed_value)
+        return signer.unsign(
+            signed_value,
+            max_age=settings.OPEN_DISCUSSIONS_UNSUBSCRIBE_TOKEN_MAX_AGE_SECONDS
```

Comment:
just a thought, but we may want to set this to either this `MAX_AGE` variable or the length of the subscription period (daily, weekly, etc) whichever is shorter, so that a daily subscriber would be getting a new token every day (and the one from yesterday would be invalidated). I guess that's a bit tricky though, because then we'd need to unsign the value, figure out the username, check what their email notification length is...yeah, lets not do that, nvm, haha.